### PR TITLE
Add maven wrapper

### PR DESCRIPTION
**Pull request by geichelberger**
The Maven wrapper is handy if you need to install the necessary Maven version but cannot upgrade/downgrade or don't want to install Maven at all.

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

File: .mvn/wrapper/maven-wrapper.jar
Comment (lkiesow) [inline .mvn/wrapper/maven-wrapper.jar:1]: Looking at the script, this file will be downloaded automatically by the wrapper script if you use it, isn't it? So, we probably don't need to include this?

File: mvnw.cmd [hunk 1]
```diff
@@ -0,0 +1,205 @@
+@REM ----------------------------------------------------------------------------
```
Comment (lkiesow) [inline mvnw.cmd]: We don't support Windows and we know that Opencast won't build on Windows. We probably don't need this file, do we?

**Review by lkiesow — CHANGES_REQUESTED** [review]
Should we also include additional patterns in `.gitignore`?

File: .mvn/wrapper/maven-wrapper.jar
Comment (geichelberger) [reply .mvn/wrapper/maven-wrapper.jar:1]: If the jar file is not found, the script tries to download it. Maven wrapper supports different types of integration.
The default one is with the jar file, and then there is a source, script, and only script option.
https://maven.apache.org/wrapper/#usage-without-binary-jar

File: mvnw.cmd (hunk 1 above)
Comment (geichelberger) [reply mvnw.cmd]: I will remove the file, I almost removed it from the staging area.

**Comment by mtneug** [comment]
Is it out of scope to "grep-and-replace" `mvn` with `./mvnw`, especially in the documentation?

**Comment by gregorydlogan** [comment]
@mtneug agree with you, and easy enough to do in a spare minute.  Since I made the changes someone else (you?) should probably review.  Basic ./mvnw work seems to work just fine in my testing.

File: docs/guides/developer/docs/participate/infrastructure/maven-repository.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ There is a preconfigured Docker image for a Nexus server set-up for Opencast. To
 follow these steps:
 
     docker run \
-        --name mvncache \
+        --name ./mvnwcache \
```
Comment (mtneug) [inline docs/guides/developer/docs/participate/infrastructure/maven-repository.md]: @gregorydlogan too much "grep-and-replace" 😉 

```suggestion
        --name mvncache \
```

**Review by mtneug — CHANGES_REQUESTED** [review]

**Review by mtneug — APPROVED** [review]

File: .mvn/wrapper/maven-wrapper.jar
Comment (mtneug) [reply .mvn/wrapper/maven-wrapper.jar:1]: If including the JAR is the default, then this is fine by me. I have only used it in that way as well.

**Pull request merged**